software/firmware/europi.py

```diff
@@ -472,21 +474,57 @@ def __init__(
                 )
         super().__init__(self.width, self.height, i2c)
 
-    def centre_text(self, text):
-        """Split the provided text across 3 lines of display."""
+    def writer(self, font):
+        """Returns the large font writer for the specified font."""
+        n = font.__name__
+        if n not in self.writers:
+            self.writers[n] = Writer(self, font)
+        return self.writers[n]
+
+    def text_len(self, s, font=None):
+        """Returns the length of the string s in pixels."""
+        if font is None:
+            return len(s) * 8
+        else:
+            return self.writer(font).string_len(s)
+
+    def text(self, s, x, y, c=1, font=None):
+        """Display the string s using the x, y coordinates as the upper-left corner of the text.
+        When using a custom font, the text is not displayed if it does not fit in the display.

```

Comment:
It's a personal preference. I prefer to make sure, during the design phase, that no text will exceed the limits of the display rather than having text that is truncated.

Reply:
I think that might be confusing to script authors and users. Absence of text could be caused by a number of problems, but when seeing your text cut off, it's immediately obvious what the problem is.

Reply:
Also, always showing the text can help with the initial layout of the display. May times I have thrown some text on the screen to see how it looked, where it got cut off, test out ideas, etc. If I had to fit it perfectly before being able to see  it, the whole process would have been a lot more cumbersome.

Reply:
You have convinced me that a clipped text is better than no text at all. I will update the code to go in that direction.